### DISCUSSION TITLE
ADIF Export of Search Results (Advanced Search)

### DISCUSSION THREAD

**Discussion by dg9vh** [magicbug/Cloudlog#805] [Ideas / Feature Requests]
I often has the wish to export explizit a set of QSOs (for example one special mode or a defined time) into ADIF for processing the QSOs with some other software.

So would it be possible to add just a "Export as ADIF"-Button to the Search Results-Table like sketched below:
![image](https://user-images.githubusercontent.com/13950650/104096176-30f98700-529b-11eb-83d3-380e940e0fef.png)

How about this idea?

73 de Kim
DG9VH

**Comment by magicbug** [comment]
No objection to the idea, you'd just have to grab the search query to run the export but that wouldn't be too impossible

**Comment by dg9vh** [comment]
That's what I am thinking, too... Maybe we could also work with the result-set at this time... but if I am the only one wanting this it would make no sense to put some power into this.

**Comment by ea5wa** [comment]
Could be useful... I need it to export my qsos with AM1SAT

**Reply by AndreasK79** [reply]
@ea5wa and @dg9vh try out the latest master. After tinkering away, I managed to get it to export to an ADIF-file without too mange changes.

**Comment by dg9vh** [comment]
how to use the export in latest master? Can't see the functionality in WebUI...

**Comment by dg9vh** [comment]
ok, found - was somehow a browser-cache-problem... after cleaning cache and reloading, it was fixed :-)

**Comment by dg9vh** [comment]
Tested functional :-)

**Reply by AndreasK79** [reply]
@dg9vh Great. I couldn't really understand why it wasn't there. 

The biggest problem was to get the javascript to trigger the browser to download a file. Getting the search result was the easy part.

**Comment by dg9vh** [comment]
:-) nice done... If we meet some day, your beer is safe :-)

**Reply by AndreasK79** [reply]
@dg9vh thanks. Who knows what the future will bring :)

I have an idea to tackle the saving of searches as well.
1. Make a table to store the compiled query.
2. When page loads, it loads buttons with stored queries so that you can run these.
3. GUI for managing the queries.
4. Somehow be able to export these results.

**Comment by dg9vh** [comment]
Also good idea :-) Exporting results via adif is working already :-) so
only storing and rebuilding the query is to be done :-)

Am Do., 23. Sept. 2021 um 20:54 Uhr schrieb Andreas Kristiansen <
***@***.***>:

> @dg9vh <https://github.com/dg9vh> thanks. Who knows what the future will
> bring :)
>
> I have an idea to tackle the saving of searches as well.
>
>    1. Make a table to store the compiled query.
>    2. When page loads, it loads buttons with stored queries so that you
>    can run these.
>    3. GUI for managing the queries.
>    4. Somehow be able to export these results.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/magicbug/Cloudlog/discussions/805#discussioncomment-1376350>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADKN5ORNISUM4PDZPLK7JE3UDNZWRANCNFSM4V3VVWPA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Amateurfunk ist das, was uns verbindet! Bleiben wir in Kontakt!
Ortsverband Q03, Neunkirchen

**Reply by AndreasK79** [reply]
Not sure how you would rebuild a compiled query. It's easier to just run the compiled query instead. 

The only downside would be that you can't change the stored query. 

I think building something to rebuild the query, would be more work alone, than building everything the way described above. 

I might be wrong, but time will tell.

**Comment by dg9vh** [comment]
Yes, you are right... I only see the way to store the created sql-query ..
but maybe in a second step (maybe years later, hi) we can have a SQL to
whatever-parser... or other idea is to convert the thing as a JSON-string
and store this... but actually no brain free for thinking about this in
detail... to much traffic in QRL... it's just before semester-starting at
our university and we have so many students to get them ready for the new
semester :-)

Am Do., 23. Sept. 2021 um 21:21 Uhr schrieb Andreas Kristiansen <
***@***.***>:

> Not sure how you would rebuild a compiled query. It's easier to just run
> the compiled query instead.
>
> The only downside would be that you can't change the stored query.
>
> I think building something to rebuild the query, would be more work alone,
> than building everything the way described above.
>
> I might be wrong, but time will tell.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/magicbug/Cloudlog/discussions/805#discussioncomment-1376461>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADKN5OSWNSZWIF7PWG5RNA3UDN42FANCNFSM4V3VVWPA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Amateurfunk ist das, was uns verbindet! Bleiben wir in Kontakt!
Ortsverband Q03, Neunkirchen

**Reply by AndreasK79** [reply]
I can imagine that it's busy this time a year at the university. 

I'll see if I can make something regarding saving queries. I have it pretty much figured it out in my head. Time will tell if the code agrees with me :)

**Comment by ea5wa** [comment]
I still don't see how can I export only the QSOs with AM1SAT....

**Reply by T012m3n7oR** [reply]
The export button will be there before you start building your query.  If you don't see it you're not on the latest build.
The search button can be clicked with an empty search to get to the advanced search page.

**Comment by dg9vh** [comment]
Just build your search and then click on Export after searching and
checking the result is correct:

![image](https://user-images.githubusercontent.com/13950650/134573678-e18fffe8-8280-4503-8a51-42f44b6f961d.png)

**Comment by magicbug** [comment]
@AndreasK79 Might be worth hiding the export till after a search has been made as I can see people trying to press it before hand

**Reply by AndreasK79** [reply]
@magicbug That can be done, but since it really has no relation to the search-button, I didn't bother. I can refine this, but I was so happy get it to work, I wanted to get it out there :)
